crates/nu-command/src/formats/to/nuon.rs

```diff
@@ -172,14 +172,28 @@ fn value_to_string(v: &Value, span: Span) -> Result<String, ShellError> {
 
 fn value_to_string_without_quotes(v: &Value, span: Span) -> Result<String, ShellError> {
     match v {
-        Value::String { val, .. } => Ok({
-            let mut quoted = escape_quote_string(val);
-            if !needs_quotes(val) {
-                quoted.remove(0);
-                quoted.pop();
+        Value::String { val, .. } => {
+            if val.contains('\t') {
+                return Err(ShellError::UnsupportedInput(
+                    "tab character cannot be converted".to_string(),
+                    span,
+                ));
             }
-            quoted
-        }),
+            if val.contains('\n') {
+                return Err(ShellError::UnsupportedInput(
+                    "newline character cannot be converted".to_string(),
+                    span,
+                ));
```

Comment:
Do we have code already to escape them to `\t` and `\n` respectively?

Reply:
I agree, the allowed / disallowed characters are already handled by the parser. If we want to disallow some characters in column names, let's do it in the parser. Here, we should just check for characters that break the `to nuon | from nuon` roundtrip and wrap them in quotes.